subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/FixedAgeOldestCacheCleanup.java

```diff
@@ -38,10 +38,11 @@ public FixedAgeOldestCacheCleanup(long ageInDays) {
         this.minimumTimestamp = Math.max(0, System.currentTimeMillis() - TimeUnit.DAYS.toMillis(ageInDays));
     }
 
+    @Override
     protected List<File> findFilesToDelete(final PersistentCache persistentCache, File[] filesEligibleForCleanup) {
         LOGGER.info("{} remove files older than {}.", persistentCache, new Date(minimumTimestamp));
 
-        final List<File> filesForDeletion = Lists.newArrayList();
+        List<File> filesForDeletion = Lists.newArrayListWithCapacity(filesEligibleForCleanup.length);
```

Comment:
I hope we evict fewer files than all the files in the cache. I guess we should go with `newArrayList` here instead of possibly allocating a too big array.

Reply:
This is a temporary list that will get discarded immediately, so size doesn't matter much. OTOH, we avoid having to re-allocate the array ever.